test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -834,60 +834,74 @@ UCS_TEST_P(test_ucp_sockaddr, compare_cm_and_wireup_configs) {
 UCP_INSTANTIATE_ALL_TEST_CASE(test_ucp_sockaddr)
 
 
-class test_ucp_sockaddr_different_tl_rsc : public test_ucp_sockaddr {
+class test_ucp_sockaddr_different_tl_rsc : public test_ucp_sockaddr
+{
+public:
+    static void get_test_variants(std::vector<ucp_test_variant>& variants)
+    {
+        uint64_t features = UCP_FEATURE_STREAM | UCP_FEATURE_TAG;
+        test_ucp_sockaddr::get_test_variants_mt(variants, features,
+                                                UNSET_SELF_DEVICES,
+                                                "unset_self_devices");
+        test_ucp_sockaddr::get_test_variants_mt(variants, features,
+                                                UNSET_SHM_DEVICES,
+                                                "unset_shm_devices");
+        test_ucp_sockaddr::get_test_variants_mt(variants, features,
+                                                UNSET_SELF_DEVICES |
+                                                UNSET_SHM_DEVICES,
+                                                "unset_self_shm_devices");
+    }
+
 protected:
-    void init() {
+    enum {
+        UNSET_SELF_DEVICES = UCS_BIT(0),
+        UNSET_SHM_DEVICES  = UCS_BIT(1)
+    };
+
+    void init()
+    {
         m_err_count = 0;
         get_sockaddr();
         test_base::init();
-
-        /* entities will be created in a test */
-    }
-
-    void cleanup() {
-        /* cleanup will be done in a test */
+        // entities will be created in a test
     }
 
     // create entities with different set of MDs and TL resources on a client
     // and on a server to test non-homogeneous setups
-    void do_test(bool self_devices, bool shm_devices) {
-        if (self_devices) {
+    void do_test(bool unset_self_devices, bool unset_shm_devices)
```

Comment:
instead of passing params - can use get_variant_value() here

Reply:
done